File: Crc8Calculator.c
#include<stdio.h>

int Crc_CalculateCRC8Runtime(short Crc_Length, short Crc_Data[])
{
    short Crc_DataAddr = 0;
    short Crc_LoopCounter = 0;
    short XOR_Value = 0xFF;
    short Crc_poly = 0x1D;
    short init_CRC_Value = 0xFF;
    short Crc_Value;

    init_CRC_Value = 0xFF;
    Crc_Value = init_CRC_Value;
    
    for (Crc_DataAddr = 0; Crc_DataAddr < Crc_Length; Crc_DataAddr++)
    {
        Crc_Value ^= Crc_Data[Crc_DataAddr]; 
        
        for (Crc_LoopCounter = 0; Crc_LoopCounter < 8; Crc_LoopCounter++)
        {
            if ((Crc_Value & 0x80u) > 0)
            {
                Crc_Value = ((Crc_Value << 1)) ^ Crc_poly;
            }
            else
            {
                Crc_Value = (Crc_Value << 1);
            }
        }
    }
    Crc_Value ^= XOR_Value;
    return Crc_Value;
}

int main(void)
{
    short crcRaw[] = {0x70, 0x00, 0x0F, 0xA0, 0xFA, 0x0A, 0x0C}; // 0xD8
    short crcValue;
    crcValue = Crc_CalculateCRC8Runtime(sizeof(crcRaw) / sizeof(crcRaw[0]), crcRaw);
    // sizeof(crcRaw)： 数组总长度的总字节数 / sizeof(crcRaw[0])： 数组单个元素的字节长度
    printf("0x%x", crcValue & 0xFF); // 取数值后2位

    return 0;
}

File: Crc8Calculator.py
# encoding: utf-8

CrcRaw = [0x70, 0x00, 0x0F, 0xA0, 0xFA, 0x0A, 0x0C] # 0xD8
# CrcRaw = [0x70, 0xA0] # 0x34

def Crc_CalculateCRC8Runtime(Crc_Length, Crc_Data):

    # This value is XORed to the final register value before returned as the official checksum.
    XOR_Value = 0xFF # Not 0x0

    # Defines the generator polynomial which is used for the CRC algorithm.
    Crc_Poly = 0x1D # Autosar profile 1, #define CRC_POLYNOMIAL_8  (0x1Du)  10000 0000

    # Defines the start condition for the CRC algorithm.
    init_CRC_Value = 0xFF # Default value is not 0x0 absolutely. 
    
    # Checksum value for return.
    Crc_Value = init_CRC_Value
    
    # #10 Perform CRC calculation for each int in Crc_Data
    for Crc_DataAddr in range(0, Crc_Length, 1):

        Crc_Value ^= Crc_Data[Crc_DataAddr]

        # #20 XOR next int of Crc_Data with current CRC value. 
        # This is equivalent to calculating CRC value of concatenated ints.
        for Crc_LoopCounter in range(0, 8, 1):
            if ((Crc_Value & int(0x80)) > 0): 
                # #35 If MSB is 1, CRC value is XORed with polynomial
                Crc_Value = ((Crc_Value << 1)) ^ Crc_Poly
            else:
                Crc_Value = (Crc_Value << 1)

    Crc_Value ^= XOR_Value
    return Crc_Value

if __name__ == '__main__':
    CrcReturn = Crc_CalculateCRC8Runtime(len(CrcRaw), CrcRaw)
    # 整型位宽64bit，高位参与了运算，取数值后2位
    print("0x%x"%(CrcReturn & 0xFF))